### ISSUE TITLE
Has the August 16th, 2018 endpoint migration been adressed?

### ISSUE THREAD

**Issue by GarkGarcia** [ttezel/twit#458]
Hey, I have a Twitter bot which was working fine, but I noticed it stopped working a while ago (I'm getting `Error: Bad Twitter streaming request: 410`).

I've researched about it and I found out [many legacy endpoints and services are getting switched off](https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/migration/introduction) right about today. Has this been addressed yet?

**Comment by super-ienien** [comment]
Hi,

everything is described here : https://twittercommunity.com/t/details-and-what-to-expect-from-the-api-deprecations-this-week-on-august-16-2018/110746

There will be a period of flickering and then twitter will shutdown the user streams on 23rd of august.

**Comment by GarkGarcia** [comment]
@super-ienien Yep. According to the [developer.twitter link I posted here](https://developer.twitter.com/en/docs/accounts-and-users/subscribe-account-activity/migration/introduction):

> Site Streams and User Streams will be fully retired on August 23rd

I'm asking if the Twit API has already been updated to addressed those changes (or if it will be updated soon). Will the `Twit.stream()` function be deprecated then? How's Twit going to change on regards to this update?

**Comment by super-ienien** [comment]
Sorry for my misunderstanding.

Twitter announced thoses changes 6 months ago. I think if the author of Twit had planned to do something he would have done it already.

**Comment by vineeth030** [comment]
I also have a twitter Bot using twit which is giving same error message. I got this package, [twitter-webhooks](https://github.com/super-ienien/twitter-webhooks) to use Activity API of twitter. If anyone here have a better idea please let me know.

**Comment by maziyarpanahi** [comment]
By addressing you mean only two things possible:

1- Removing those functions since they are being deprecated: They haven't entirely been shut down. So removing them won't be fair to other people who are still using them. Also, just a change in the documentation to say these two are being deprecated is enough. No need to remove anything in the code.

2- Implementing the new User Stream and Site Stream (Sandbox - Premium APIs): This requires Developer Approval process (from 1 week all the way to 5 weeks or more!). You have to be approved and pass to see how these APIs really work and be able to test them. It is not as simple as creating an application before. I understand if the maintainers and contributors are taking their time until everything is already settled or in the queue to be approved.

> Will the Twit.stream() function be deprecated then?
If you mean Twitter Public Streaming APIs, the answer is yes but not soon. It has been in the roadmap for a while, but they haven't moved on to it yet as the other APIs impact their business more than Streaming APIs.

My suggestion, if your app entirely relies on Site Stream and User Stream then you should use other libraries.

**Comment by vineeth030** [comment]
Thank you @maziyarpanahi

**Comment by GarkGarcia** [comment]
@maziyarpanahi I thought about adapting those function to make use of `webhooks` or something along those lines, but I don't really know how that would work. Maybe deprecating those functions and creating new ones - based on `webhooks` technology - would be a good idea too.

Sure, I understand if the developers take their time.

As for myself, I think I'm gonna dig a little bit into the Twitter API and see if I can replicate everything just how it was.

**Comment by vineeth030** [comment]
@GarkGarcia I finally referred their official code sample [https://github.com/twitterdev/account-activity-dashboard](https://github.com/twitterdev/account-activity-dashboard) That helped me.

**Comment by super-ienien** [comment]
@vineeth030 Did you try twitter-webhooks module ?

I’m the author, if you tried i’m Interested by your feedbacks and reasons you didn’t use it. It would help me to make improvements on it. Thanks ;)

**Comment by super-ienien** [comment]
@GarkGarcia I was in your case and that's why I did [twitter-webhooks](https://github.com/super-ienien/twitter-webhooks). It implements all you need to receive twitter user events like user stream.

But the new User Activity API is much more restrictive than the User Stream API... First you need a server or an App running on a Machine reachable by Twitter via an URL, second you need HTTPS and third your App needs to be approved by Twitter.

**Comment by vineeth030** [comment]
@super-ienien I only had 8 hours to make my Bot working again. So when I got stuck in some errors. I looked for alternative. But I did referred your module code a lot. It did help me along the way. When I get free time I``ll get back to it.  That`s a great effort, creating that module.

**Comment by super-ienien** [comment]
@vineeth030 Ok thanks ;)

**Comment by alveshelio** [comment]
Hi guys,

I'm working on a small project and one of the requirements is to get tweets from a specific user. @vineeth030 Do you think you could share what you did as work around?

@super-ienien I've checked your module and in userActivity, it seems that we can only subscribe to a user if we know the userId and the accessToken and accessTokenSecret, it means your own account. Is there a way to get tweets from another account and display them on a dashboard?

Thank you

**Comment by super-ienien** [comment]
@alveshelio Account Activity API is not what you need...

You just need this : https://developer.twitter.com/en/docs/tweets/filter-realtime/guides/basic-stream-parameters

This feature is supported by Twit. Anyway, you'll still need to get the userId of the users you wan't to follow, you can retrieve them easily by using the Twitter's REST API. Just look into the doc.

**Comment by tracycollins** [comment]
How should I handle Error: Bad Twitter streaming request: 410?  

Right now, it crashes my app.  I'm already listening for and handling stream errors already.

**Comment by vineeth030** [comment]
@tracycollins stream APIs has been deprecated. Use account activity API if your requirement is to listen to direct message or tweets.

**Comment by dandv** [comment]
Not sure if this library is actively maintained. [twitter-lite](https://github.com/draftbit/twitter-lite/) is, and we're happy to look into this if you test the functionality and file an issue.
